### PR TITLE
feat: CRW-2818 crwctl is now dsc; codeready...

### PR DESCRIPTION
**Pull request by nickboldt**
### What does this PR do?

feat: CRW-2818 crwctl is now dsc; codeready and codeready-workspaces is now devspaces; refactor all the things

Change-Id: Ibd646873249c58c0109f1e7dea1d25a11e0595a7
Signed-off-by: nickboldt <nboldt@redhat.com>

### Screenshot/screencast of this PR
N/A

### What issues does this PR fix or reference?
N/A (or see commit message above for issue number)

### How to test this PR?
N/A

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works, if one exists](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections What issues does this PR fix or reference and How to test this PR completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by nickboldt** [comment]
```
$➔ yarn && npx oclif-dev pack
yarn install v1.22.5
[1/5] Validating package.json...
[2/5] Resolving packages...
success Already up-to-date.
$ npm run -s postinstall-repositories && npm run -s postinstall-operator && npm run -s postinstall-cleanup
yarn upgrade v1.22.5
[1/5] Validating package.json...
[2/5] Resolving packages...
[3/5] Fetching packages...
info fsevents@2.1.3: The platform "linux" is incompatible with this module.
info "fsevents@2.1.3" is an optional dependency and failed compatibility check. Excluding it from installation.
[4/5] Linking dependencies...
warning "eslint-config-oclif > eslint-config-xo-space@0.20.0" has incorrect peer dependency "eslint@^5.3.0".
[5/5] Rebuilding all packages...
success Saved lockfile.
success Saved 1 new dependency.
info Direct dependencies
└─ devspaces-operator@0.0.0
info All dependencies
└─ devspaces-operator@0.0.0
Done in 74.12s.
/home/nboldt/5-Che/1.github_upstream__manual-sync_to_pkgs.devel/ds-chectl/prepare-devspaces-operator-templates.js:47
        throw new Error("Unable to prepare devspaces-operator templates")
        ^

Error: Unable to prepare devspaces-operator templates
    at prepareTemplates (/home/nboldt/5-Che/1.github_upstream__manual-sync_to_pkgs.devel/ds-chectl/prepare-devspaces-operator-templates.js:47:15)
    at Object.<anonymous> (/home/nboldt/5-Che/1.github_upstream__manual-sync_to_pkgs.devel/ds-chectl/prepare-devspaces-operator-templates.js:52:1)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:17:47
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

File: build/scripts/sync.sh [hunk 1]
```diff
@@ -145,19 +145,19 @@ pushd "${TARGETDIR}" >/dev/null
...
+# @since 2.11: CRW-2150 - sources have moved to https://github.com/redhat-developer/devspaces-images/tree/crw-2-rhel-8/devspaces-operator
 # Update prepare-che-operator-templates.js
 pushd "${TARGETDIR}" >/dev/null
 	while IFS= read -r -d '' d; do
 		echo "[INFO] Convert ${d} - use subfolder"
 		sed -i "${TARGETDIR}/${d}" -r \
-			-e "s#'node_modules', 'codeready-workspaces-operator'#'node_modules', 'codeready-workspaces-operator', 'codeready-workspaces-operator'#" \
-			-e "s#'templates', 'codeready-operator'#'templates', 'codeready-workspaces-operator'#"
+			-e "s#'node_modules', 'devspaces-operator'#'node_modules', 'devspaces-operator', 'devspaces-operator'#" \
+			-e "s#'templates', 'devspaces-operator'#'templates', 'devspaces-operator'#"
```
Comment (amisevsk) [inline build/scripts/sync.sh:155]: This replacement won't work until the directories in https://github.com/redhat-developer/devspaces-images#crw-2-rhel-8 are updated to be `devspaces-operator`:

```
ls node_modules/devspaces-operator/
total 32K
drwxr-xr-x. 1 amisevsk  478 Mar 16 16:36 codeready-workspaces
drwxr-xr-x. 1 amisevsk  326 Mar 16 16:36 codeready-workspaces-configbump
drwxr-xr-x. 1 amisevsk  610 Mar 16 16:36 codeready-workspaces-dashboard
drwxr-xr-x. 1 amisevsk  444 Mar 16 16:36 codeready-workspaces-devfileregistry
drwxr-xr-x. 1 amisevsk  304 Mar 16 16:36 codeready-workspaces-idea
...
```

File: package.json [hunk 2]
```diff
@@ -80,7 +80,7 @@
     "ansi-colors": "4.1.1",
     "axios": "^0.21.1",
     "cli-ux": "^6.0.9",
-    "codeready-workspaces-operator": "https://github.com/redhat-developer/codeready-workspaces-images#crw-2-rhel-8",
+    "devspaces-operator": "https://github.com/redhat-developer/devspaces-images#crw-2-rhel-8",
```
Comment (amisevsk) [inline package.json:83]:
```suggestion
    "devspaces-operator": "https://github.com/redhat-developer/devspaces-images#devspaces-3-rhel-8",
```

File: build/scripts/sync.sh (hunk 1 above)
Comment (amisevsk) [reply build/scripts/sync.sh:155]: Just realized the branch name -- see below

**Comment by nickboldt** [comment]
fixed, thanks!

```
oclif-dev: packing tarball from ./tmp/dsc to ./dist/channels/CI-redhat/dsc-linux-x64.tar.gz
oclif-dev: building target dsc-linux-x64.tar.gz
oclif-dev: downloading node-v16.13.2-linux-x64
node-v16.13.2-linux-x64.tar.xz: OK
oclif-dev: extracting node-v16.13.2-linux-x64
oclif-dev: packing tarball from ./tmp/linux-x64/dsc to ./dist/channels/CI-redhat/dsc-linux-x64.tar.gz
oclif-dev: building target dsc-linux-arm.tar.gz
oclif-dev: downloading node-v16.13.2-linux-armv7l
node-v16.13.2-linux-armv7l.tar.xz: OK
oclif-dev: extracting node-v16.13.2-linux-armv7l
oclif-dev: packing tarball from ./tmp/linux-arm/dsc to ./dist/channels/CI-redhat/dsc-linux-arm.tar.gz
oclif-dev: building target dsc-win32-x64.tar.gz
oclif-dev: downloading node-v16.13.2-win-x64
node-v16.13.2-win-x64.7z: OK
oclif-dev: extracting node-v16.13.2-win-x64
oclif-dev: packing tarball from ./tmp/win32-x64/dsc to ./dist/channels/CI-redhat/dsc-win32-x64.tar.gz
oclif-dev: building target dsc-win32-x86.tar.gz
oclif-dev: downloading node-v16.13.2-win-x86
node-v16.13.2-win-x86.7z: OK
oclif-dev: extracting node-v16.13.2-win-x86
oclif-dev: packing tarball from ./tmp/win32-x86/dsc to ./dist/channels/CI-redhat/dsc-win32-x86.tar.gz
oclif-dev: building target dsc-darwin-x64.tar.gz
oclif-dev: downloading node-v16.13.2-darwin-x64
node-v16.13.2-darwin-x64.tar.xz: OK
oclif-dev: extracting node-v16.13.2-darwin-x64
oclif-dev: packing tarball from ./tmp/darwin-x64/dsc to ./dist/channels/CI-redhat/dsc-darwin-x64.tar.gz
```

**Pull request merged**